### PR TITLE
Fix for issues with spatial fields

### LINKED ISSUE #14067: IS NULL search operator produces no WHERE SQL with spatial fields
### Steps to reproduce
1.
```
CREATE TABLE `t` (
 `point` POINT DEFAULT NULL
) ENGINE=InnoDB;

INSERT INTO `t` (`point`) VALUES (NULL), (PointFromText('POINT(1 1)'));
```
2. Go to Search tab on phpmyadmin
3. Change point operator to "IS NULL" (or "IS NOT NULL")
4. Click Go

### Expected behaviour
Results should show 1 entry for the NULL point

### Actual behaviour
No WHERE clause is added to SQL so whole table is returned

### Server configuration
**Operating system**:
CentOS
**Web server:**
Apache/2.4.6
**Database:**
10.2.12-MariaDB
**PHP version:**
5.6.33
**phpMyAdmin version:**
4.7.7

### LINKED ISSUE #14068: SPATIAL fields with no unique index/autoincrement error when clicking
I discovered this when looking at #14067; maybe it's related or a duplicate.

### Steps to reproduce
1. Start with a table containing no unique index (which will cause grid editing to fall back to our best effort to isolate the correct row):

```
CREATE TABLE `ptt` (
 `pt` POINT DEFAULT NULL
) ENGINE=InnoDB;

INSERT INTO `ptt` (`pt`) VALUES (NULL), (PointFromText('POINT(1 1)'));
```

2. Browse the table and click the data, which should offer to download the row (or maybe let me edit it).

### Expected behaviour
I should either be offered to download the data or grid edit.

### Actual behaviour
An error is shown without any of the phpMyAdmin menu bar, navigation pane, etc:
```
Error

SQL query: DocumentationEdit Edit

SELECT `pt` FROM `ptt` WHERE ;

MySQL said: Documentation
MySQL returned an empty result set (i.e. zero rows).
```

### Server configuration
**Operating system**:
Debian
**Web server:**
Apache
**Database:**
MariaDB 10.1.26
**PHP version:**
7.0.27
**phpMyAdmin version:**
master

### Client configuration
**Browser:**
Firefox
**Operating system:**
MacOS

### LINKED ISSUE #14121: View spatial fields as WKT doesn't work
### Steps to reproduce
1. make a table with spatial fields
```
`CREATE TABLE t (
 point POINT DEFAULT NULL
) ENGINE=InnoDB;

INSERT INTO `t` (`point`) VALUES (NULL), (PointFromText('POINT(1 1)'));`
```
2. View the table contents as wkt(well known text) in the options menu.

### Expected behaviour
Blank values in the corresponding cell.

### Actual behaviour
Should display the cell as wkt

### Server configuration
**Operating system**:
ubuntu 16.04

**Web server:**
Apache/2.4.29 (Unix)

**Database:**
10.1.30-MariaDB

**PHP version:**
7.1.14

**phpMyAdmin version:**
4.8.0-dev

### PR DESCRIPTION
**Pull request by aroralakshya**
Fixes the following issues:
1. Fixes #14068 SPATIAL fields with no unique index/autoincrement error when clicking
	This error occurs when there is no unique column to uniquely identify a tuple. I've disabled the geomtery view and made view as wkt the default view in such cases.
2. Fixes #14121 View spatial fields as WKT doesn't work
3. Fixes #14067 IS NULL search operator produces no WHERE SQL with spatial fields
Signed-Off-By: Lakshya arora <arora.lakshya123@gmail.com>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

### CODE REVIEW

**Comment by nulll-pointer** [comment]
How to add test cases to improve coverage??

**Comment by mauriciofauth** [comment]
Hello, @nulll-pointer. Thanks for your contribution!
Each class has its own unit test class. These classes are found in the `test/classes/` directory.
Run the `vendor/bin/phpunit` command to test.

File: libraries/classes/Display/Results.php [hunk 1]
```diff
@@ -1680,13 +1680,14 @@ private function _getOptionsBlock()
...
             'unique_id' => $this->__get('unique_id'),
-            'geo_option' => $_SESSION['tmpval']['geoOption'],
+            'geo_option' => ($_SESSION['tmpval']['possible_as_geomtery'] ? $_SESSION['tmpval']['geoOption'] : (self::GEOMETRY_DISP_WKT)),
             'hide_transformation' => $_SESSION['tmpval']['hide_transformation'],
             'display_blob' => $_SESSION['tmpval']['display_blob'],
             'display_binary' => $_SESSION['tmpval']['display_binary'],
             'relational_display' => $_SESSION['tmpval']['relational_display'],
             'displaywork' => $GLOBALS['cfgRelation']['displaywork'],
             'relwork' => $GLOBALS['cfgRelation']['relwork'],
+            'possible_as_geometry' => $_SESSION['tmpval']['possible_as_geomtery'],
```
Comment (mauriciofauth) [inline libraries/classes/Display/Results.php]: This looks wrong. Are you sure it's correct?

File: libraries/classes/Display/Results.php [hunk 2]
```diff
@@ -1687,6 +1692,7 @@ private function _getOptionsBlock()
...
```
Comment (mauriciofauth) [inline libraries/classes/Display/Results.php]: There's still a typo in `possible_as_geomtery`.

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**